### DISCUSSION TITLE
New value for Combobox

### DISCUSSION THREAD

**Discussion by KingMario** [vmware/clarity#6583] [General]
The combobox is a great component for search and select. While in some cases, the user might need to enter a new value besides the options.

Could you please make it possible to set the combobox allow new value and also set the maxlength for the new value.

Thanks.

**Comment by Jinnie** [comment]
In general, this is a nice idea, but how will you discern between selecting an item from the dropdown (with Enter) and creating a new item (Enter again?).
Even now you can try to wire a demo, by using a `(keyup.enter)="newItem()"` on the `clr-combobox` element and then get the `ClrCombobox.searchText` property of the combobox. But it's really hard to clean up the corner cases and even harder to consider the result accessible. The combobox is a component with very complex accessibility scenario, which is even now suffering certain problems. By making it even more complex, the accessibility plane will get even larger, with more chance for a disabled user to get confused and lost.
So, though it may be a neat approach, we'd still recommend to use additional control for the input of new element.
